invokeai/app/api/routers/images.py

```diff
@@ -228,20 +224,33 @@ async def get_image_urls(
     response_model=PaginatedResults[ImageDTO],
 )
 async def list_images_with_metadata(
-    image_type: ImageType = Query(description="The type of images to list"),
-    image_category: ImageCategory = Query(description="The kind of images to list"),
-    page: int = Query(default=0, description="The page of image metadata to get"),
-    per_page: int = Query(
-        default=10, description="The number of image metadata per page"
+    image_origin: Optional[ResourceOrigin] = Query(
+        default=None, description="The origin of images to list"
+    ),
+    include_categories: Optional[list[ImageCategory]] = Query(
+        default=None, description="The categories of image to include"
     ),
+    exclude_categories: Optional[list[ImageCategory]] = Query(
+        default=None, description="The categories of image to exclude"
+    ),
```

Comment:
The intention is to make the fetch parameters simple and resistant to changes in categories. 

For example, getting all categories except `user` is exclude `user` instead of manually listing all categories you do want. If the categories changed, we'd need to update the query. 

I suppose categories aren't going to change very often though, so it could be reduced to just the inclusive query.

Reply:
I removed the exclude and renamed the inclusive filter to just `categories`.